src/kbmod/filters/sigma_g_filter.py

```diff
@@ -183,16 +130,45 @@ def compute_clipped_sigma_g_matrix(self, lh):
             A N x T matrix of Booleans indicating if each point is valid (True)
             or has been filtered (False).
         """
-        # We need to
-        os.environ["XLA_PYTHON_CLIENT_PREALLOCATE"] = "false"
-        os.environ["XLA_PYTHON_CLIENT_ALLOCATOR"] = "platform"
+        # Use a GPU if one is available.
+        if torch.cuda.is_available():
+            device = torch.device("cuda")
+        else:
+            device = torch.device("cpu")
+
+        # Move the likelihood data to the device.
+        torch_lh = torch.tensor(lh, device=device, dtype=torch.float64)
+
+        # Mask out the negative values (if clip negative is true).
+        if self.clip_negative:
+            masked_lh = torch.where(torch_lh > 0.0, torch_lh, np.nan)
+        else:
+            masked_lh = torch_lh
+
+        # Compute the quantiles for each row.
+        q_values = torch.tensor(
+            [self.low_bnd / 100.0, 0.5, self.high_bnd / 100.0],
+            device=device,
+            dtype=torch.float64,
+        )
+        lower_per, median, upper_per = torch.nanquantile(masked_lh, q_values, dim=1)
+
+        # Compute the bounds for each row, enforcing a minimum gap in case all the
+        # points are identical (upper_per == lower_per).
+        delta = upper_per - lower_per
+        delta[delta < 1e-8] = 1e-8
+        nSigmaG = self.n_sigma * self.coeff * delta
+
+        num_rows = lh.shape[0]
+        num_cols = lh.shape[1]
+        lower_bnd = torch.reshape((median - nSigmaG), (num_rows, 1)).expand(-1, num_cols)
+        upper_bnd = torch.reshape((median + nSigmaG), (num_rows, 1)).expand(-1, num_cols)
 
-        jnp_lh = jnp.array(lh)
-        jpn_inds_valid = self.sigma_g_jax_fn(jnp_lh).block_until_ready()
+        # Check whether the values fall within the bounds.
+        index_valid = torch.isfinite(torch_lh) & (torch_lh < upper_bnd) & (torch_lh > lower_bnd)
 
-        # Make sure the inds_valid is on CPU as a numpy array.
-        inds_valid = np.array(jpn_inds_valid)
-        return inds_valid
+        # Return as a numpy array on the CPU.
+        return index_valid.cpu().numpy()
```

Comment:
I assume that `pytorch` will handle this gracefully when running on a machine without a gpu in the first place?

Reply:
Yes. If the tensor is already on the cpu, then it is a no-op.